prometheus/src/main/scala/filodb/prometheus/parse/AntlrParser.scala

```diff
@@ -103,9 +103,11 @@ class AntlrParser extends PromQLBaseVisitor[Object] {
 
   override def visitSubquery(ctx: PromQLParser.SubqueryContext): SubqueryClause = {
     val list = ctx.DURATION
-    val window = parseDuration(list.get(0))
-    val step = if (list.size() > 1) Some(parseDuration(list.get(1))) else None
-    SubqueryClause(window, step)
+    val windowMs = parseDuration(list.get(0))
+    val stepMs = if (list.size() > 1) Some(parseDuration(list.get(1))) else None
+    SubqueryClause(
+      Duration(windowMs / 1000, Second),
```

Comment:
since parseDuration returns a Duration with a millisecond as its unit. Do we need to encapsulate it again?
I think Duration allows operation like 1s + 1ms.

Reply:
`parseDuration` used to return a `Duration`, but it now just returns the total elapsed milliseconds in a `DURATION`-- that name is probably just confusing now, so I'll update it.

Reply:
I've changed the return type back to `Duration`.